### PR TITLE
[Translation zh-tw] Fundamentals > Primers

### PR DESCRIPTION
**Pull request by limhenry**
Chinese (Traditional) translation of:

 - src/content/zh-tw/fundamentals/primers/async-functions.md
 - src/content/zh-tw/fundamentals/primers/promises.md
 - src/content/zh-tw/fundamentals/primers/service-workers/index.md
 - src/content/zh-tw/fundamentals/primers/service-workers/lifecycle.md
 - src/content/zh-tw/fundamentals/primers/service-workers/registration.md

Note: All the articles are directly translated from zh-CN.

### CODE REVIEW

File: src/content/zh-tw/fundamentals/primers/async-functions.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        // …
+      }
+    }
+
+如果在函數定義之前使用了 `async` 關鍵字，就可以在函數內使用 `await`。
+當您 `await` 某個 Promise 時，函數暫停執行，直至該 Promise 產生結果，並且暫停並不會阻塞主線程。
+如果 Promise 執行，則會返回值。
+如果 Promise 拒絕，則會拋出拒絕的值。
+
+注：如果不熟悉 Promise，可以看一看[我們的 Promise 指南](/web/fundamentals/getting-started/primers/promises)。
```
Comment (petele) [inline src/content/zh-tw/fundamentals/primers/async-functions.md]: Don't translate [callouts](https://developers.google.com/web/resources/markdown-syntax#callouts) like `Note:` as it's a special marker for DevSite. The content should be translated, but the callout itself, shouldn't.

File: src/content/zh-tw/fundamentals/primers/async-functions.md [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+      }
+      catch (err) {
+        console.log('fetch failed', err);
+      }
+    }
+
+代碼行數雖然相同，但去掉了所有回調。這可以提高代碼的可讀性，對不太熟悉 Promise 的人而言，幫助就更大了。
+
+
+注：您 `await` 的任何內容都通過 `Promise.resolve()` 傳遞，這樣您就可以安全地 `await` 非原生 Promise。
```
Comment (petele) [inline src/content/zh-tw/fundamentals/primers/async-functions.md]: callout

File: src/content/zh-tw/fundamentals/primers/async-functions.md [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    }
+
+…調用 `foo()` 返回的 Promise 會在*拒絕*時返回 `Error('bar')`。
+
+## 示例：流式傳輸響應
+
+異步函數在更復雜示例中更有用武之地。假設我們想在流式傳輸響應的同時記錄數據塊日誌，並返回數據塊最終大小。
+
+
+注：一看到“記錄數據塊日誌”這幾個字就讓我感到不舒服。
```
Comment (petele) [inline src/content/zh-tw/fundamentals/primers/async-functions.md]: callout

File: src/content/zh-tw/fundamentals/primers/async-functions.md [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+
+### 箭頭函數
+
+    // map some URLs to json-promises
+    const jsonPromises = urls.map(async url => {
+      const response = await fetch(url);
+      return response.json();
+    });
+
+注：`array.map(func)` 不在乎我提供給它的是不是異步函數，只把它當作一個返回 Promise 的函數來看待。
```
Comment (petele) [inline src/content/zh-tw/fundamentals/primers/async-functions.md]: callout

File: src/content/zh-tw/fundamentals/primers/async-functions.md (hunk 1 above)
Comment (limhenry) [reply src/content/zh-tw/fundamentals/primers/async-functions.md]: Oops. (it's directly translated from zh-cn anyway, i will fix that later)

**Comment by limhenry** [comment]
@petele fixed :)

**Pull request merged**